cpp/include/cudf/strings/detail/strings_children.cuh

```diff
@@ -122,6 +122,62 @@ auto make_strings_children(SizeAndExecuteFunction size_and_exec_fn,
   return make_strings_children(size_and_exec_fn, strings_count, strings_count, stream, mr);
 }
 
+/**
+ * @brief Create an offsets column to be a child of a compound column
+ *
+ * This function sets the offsets values by executing scan over the sizes in the provided
+ * Iterator.
+ *
+ * The return also includes the total number of elements -- the last element value from the
+ * scan.
+ *
+ * @tparam InputIterator Used as input to scan to set the offset values
+ * @param begin The beginning of the input sequence
+ * @param end The end of the input sequence
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @return Offsets column and total elements
+ */
+template <typename InputIterator>
+std::pair<std::unique_ptr<column>, int64_t> make_offsets_child_column(
+  InputIterator begin,
+  InputIterator end,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  auto const count = static_cast<size_type>(std::distance(begin, end));
```

Comment:
Nitpick: The `count` here is the number of input string rows.  But it seems to be used as `count + 1` (i.e. as the `num_offsets`).  

Would there be value in defining `num_offsets = std::distance(begin, end) + 1` instead of `count`, and using that in place of `count+1`? 

I ask because `sizes_to_offsets(input_itr, input_iter + count + 1, ... )` is surprising on first glance.

Reply:
So I still need to pass `count` on line 159. I don't see a need to define both local variables.

Reply:
IMO `count` here sounds confusing because it doesn't tell if this is a string count of offset count. So maybe `string_count` is better here?